Problem: upgrade to keyring 2.0 fails

https://github.com/hwchen/keyring-rs#upgrading-from-v1
```
error[E0599]: no method named `set_password` found for enum `std::result::Result` in the current scope
   --> bindings/cpp/src/lib.rs:770:11
    |
770 |     entry.set_password(&infojson)?;
    |           ^^^^^^^^^^^^ method not found in `std::result::Result<keyring::Entry, keyring::Error>`
    |
note: the method `set_password` exists on the type `keyring::Entry`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-2.0.1/src/lib.rs:221:5
    |
221 |     pub fn set_password(&self, password: &str) -> Result<()> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: use the `?` operator to extract the `keyring::Entry` value, propagating a `Result::Err` value to the caller
    |
770 |     entry?.set_password(&infojson)?;
    |          +

error[E0599]: no method named `get_password` found for enum `std::result::Result` in the current scope
   --> bindings/cpp/src/lib.rs:782:26
    |
782 |     let infojson = entry.get_password()?;
    |                          ^^^^^^^^^^^^ method not found in `std::result::Result<keyring::Entry, keyring::Error>`
    |
note: the method `get_password` exists on the type `keyring::Entry`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/keyring-2.0.1/src/lib.rs:234:5
    |
234 |     pub fn get_password(&self) -> Result<String> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: use the `?` operator to extract the `keyring::Entry` value, propagating a `Result::Err` value to the caller
    |
782 |     let infojson = entry?.get_password()?;
    |                         +

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> bindings/cpp/src/lib.rs:782:9
    |
782 |     let infojson = entry.get_password()?;
    |         ^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `str`
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> bindings/cpp/src/lib.rs:782:20
    |
782 |     let infojson = entry.get_password()?;
    |                    ^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `str`
    = note: all local variables must have a statically known size
    = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> bindings/cpp/src/lib.rs:782:40
    |
782 |     let infojson = entry.get_password()?;
    |                                        ^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `str`
note: required by a bound in `std::ops::ControlFlow::Break`
   --> /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/ops/control_flow.rs:85:25
    |
    = note: required by this bound in `std::ops::ControlFlow::Break`

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `defi-wallet-core-cpp` due to 5 previous errors
```
https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/4227026836/jobs/7341085037

--- a/bindings/cpp/Cargo.toml
+++ b/bindings/cpp/Cargo.toml
@@ -20,7 +20,7 @@ tokio = { version = "1", features = ["rt"] }
 
 
 [target.'cfg(not(target_os="android"))'.dependencies]
-keyring="1.2.1"
+keyring="2"
 
 
 

--- a/bindings/cpp/src/lib.rs
+++ b/bindings/cpp/src/lib.rs
@@ -766,7 +766,7 @@ fn restore_wallet_save_to_securestorage(
     };
 
     let infojson = serde_json::to_string(&securestorageinfo)?;
-    let entry = keyring::Entry::new(&servicename, &username);
+    let entry = keyring::Entry::new(&servicename, &username)?;
     entry.set_password(&infojson)?;
     let wallet = HDWallet::recover_wallet(mnemonic, Some(password))?;
     Ok(Box::new(Wallet { wallet }))
@@ -777,7 +777,7 @@ fn restore_wallet_load_from_securestorage(
     servicename: String,
     username: String,
 ) -> Result<Box<Wallet>> {
-    let entry = keyring::Entry::new(&servicename, &username);
+    let entry = keyring::Entry::new(&servicename, &username)?;
 
     let infojson = entry.get_password()?;
     let securestorageinfo: SecureStorageWaleltInfo = serde_json::from_str(&infojson)?;

